app/controllers/users/friendships_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Users::FriendshipsController < ApplicationController
+  def create
+    unless @current_user
+      flash[:alert] = "Please log in"
+      redirect_to root_path
+      return
+    end
+    if @current_user.verify_friendship?(User.find(params[:user_id]))
+      flash[:alert] = "You are already friends!"
+      redirect_to user_path(@current_user.id)
+    elsif @current_user != User.find(params[:user_id])
+      @current_user.make_friends_with(User.find(params[:user_id]))
+      flash[:notice] = "Congrats! You are now friends with #{User.find(params[:user_id]).name}."
```

Comment:
every time we use `.find` it's another database call. We can cut this down to just one call.